.github/workflows/publish.yml

```diff
@@ -14,38 +17,81 @@ on:
 # or [0-9]+.[0-9]+.[0-9]+-rc optionally followed by digits
 
 jobs:
-  validate-tag:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Validate tag format
-        run: |
-          TAG="${{ github.event.release.tag_name }}"
-          echo "Validating tag: $TAG"
-          if [[ $TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+(-rc[0-9]*)?$ ]]; then
-            echo "Tag matches semver."
-          else
-            echo "Error: tag does not match semver"
-            exit 1
-          fi
-  publish:
-    needs: validate-tag
-    name: Build and Upload module
-    runs-on: ubuntu-latest
+  # If we use the viamrobotics/build-action@v1, it farms out compilation to Linux machines no
+  # matter which OS you're targeting (which is fine for Go and Python, but not for C++).
+  # Furthermore, these Linux machines do not cache build artifacts from one run to another. So
+  # instead, we're going to build everything on Github's action runners directly.
+  build:
+    name: Build for Unix
+    strategy:
+      matrix:
+        build_target:
+          - runs_on: ubuntu-latest
+            cache_dir: ~/.conan2
+          - runs_on: ubuntu-24.04-arm
+            cache_dir: ~/.conan2
+          - runs_on: macos-latest
+            cache_dir: ~/.conan2
+          - runs_on: macos-13
+            cache_dir: ~/.conan2
+          - runs_on: windows-2019
+            cache_dir: C:\Users\runneradmin\.conan2
+    runs-on: ${{ matrix.build_target.runs_on }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-
-      - uses: actions/cache@v3
+      - name: Restore cached libraries
+        uses: actions/cache@v3
         with:
-          path: ~/.conan2
-          key: ${{runner.os}}-conan-${{ hashFiles('**/conanfile.py') }}
+          path: ${{ matrix.build_target.cache_dir }}
+          key: ${{ matrix.build_target.runs_on }}-conan-${{ hashFiles('**/conanfile.py') }}
           restore-keys: |
-            ${{ runner.os }}-conan-
+            ${{ matrix.build_target.runs_on }}-conan-
+      - name: Setup and build
+        run: |
+          make setup
+          make module.tar.gz
```

Comment:
Could you bring this to the attention of Abe, just as an FYI, since he is the author of the github build action

Reply:
Yeah, he was interested in this earlier today, so I tagged him as a reviewer, too.